src/Optimizely/ProjectConfig.php

```diff
@@ -647,6 +648,12 @@ public function setForcedVariation($experimentKey, $userId, $variationKey)
             return false;
         }
 
+        // check for empty string Variation key
+        if (!is_null($variationKey) && !Validator::validateNonEmptyString($variationKey)) {
```

Comment:
Move all validations. 
Variation, Experiment and User to Optimizely class. 
ProjectConfig class should not perform input validation

Reply:
Leave this.